### PR TITLE
Cherrypick kubectl fix

### PR DESCRIPTION
**Pull request by jessfraz**
**What this PR does / why we need it**:

**Which issue this PR fixes** _(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR gets merged)_: fixes #

**Special notes for your reviewer**:

**Release note**:

``` release-note
NONE
```

Cherrypick of #31642 onto release-1.3 branch for upgrade tests

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31987)

### CODE REVIEW

File: test/e2e/kubectl.go [hunk 1]
```diff
@@ -495,24 +495,47 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
...
-					{"QoS Tier:", "BestEffort"},
-				}
-				checkOutput(output, requiredStrings)
-			})
+			classVersion, err := framework.KubectlVersionGTE(version.MustParse("v1.4.0-alpha"))
+			Expect(err).NotTo(HaveOccurred())
+			if classVersion {
+				forEachPod(func(pod api.Pod) {
+					output := framework.RunKubectlOrDie("describe", "pod", pod.Name, nsFlag)
+					requiredStrings := [][]string{
```
Comment (lavalamp) [inline test/e2e/kubectl.go]: OK I'm too dumb to see what changed here and why all these lines are copied. ? I guess this is a cherrypick so I have to take it verbatim though :(

Comment (jfrazelle) [inline test/e2e/kubectl.go]: This might help https://github.com/kubernetes/kubernetes/pull/31642

**Comment by dims** [comment]
Hmm, looks like someone with karma has to clear the do-not-merge and the cla:no labels.

**Comment by jfrazelle** [comment]
closing infavor of #32143

**Comment by jfrazelle** [comment]
reopening because the fix in #32143 isnt going to work for this, the server version and kubectl version are 1.4 just the tests running are 1.3 output

**Comment by jfrazelle** [comment]
@k8s-bot test this issue #26210

**Pull request merged**